lib/ex_doc/retriever.ex

```diff
@@ -330,8 +330,19 @@ defmodule ExDoc.Retriever do
   defp source_link(%{path: _, url: nil}, _line), do: nil
 
   defp source_link(source, line) do
-    source_url = Regex.replace(~r/%{path}/, source.url, source.path)
-    Regex.replace(~r/%{line}/, source_url, to_string(line))
+    source_url_pattern(source.url, source.path, to_string(line))
+  end
+
+  def source_url_pattern(source_url_pattern, path, line) do
```

Comment:
Can you please move this to `ExDoc.Utils`? Add a basic unit test and we are good to go!